libs/dal/src/repositories/preferences/preferences.repository.ts

```diff
@@ -28,13 +28,6 @@ export class PreferencesRepository extends BaseRepository<PreferencesDBModel, Pr
     return this.mapEntity(item);
   }
 
-  async delete(query: PreferencesQuery) {
-    const item = await this.findOne({ _id: query._id, _environmentId: query._environmentId });
```

Comment:
There is no need for an extra query when trying to delete a record. The number of affected rows returned by mongoose indicates if the record was found or not.